### ISSUE TITLE
Additional source directories are not put on classpath

### ISSUE THREAD

**Issue by RolKau** [redhat-developer/vscode-java#3544]
Additional source directories that are included using the build-helper-maven-plugin in a Maven project seem to not be made available to the language server, and are thus not picked up by for instance Go To Declaration and Find References.

Consider the attached example project, consisting of three leaf projects: Child1, Child2 and Child3, organized in two parent projects: ParentA and ParentB, where Child3 has a dependency on Child2, and Child2 has a dependency on Child1.

In the first iteration, stored in before/, Child1 and Child2 is organized under ParentA, and Child 3 is organized under ParentB. In the second iteration, stored in after/, Child2 is logically moved to ParentB, but to avoid changes in the project history, the files are still kept under ParentA's directory and included in ParentB using build-helper-maven-plugin.

##### Environment
- Operating System: Ubuntu Focal (20.04)
- JDK version: OpenJDK 17.0.10
- Visual Studio Code version: 1.87.2
- Java extension version: 1.28.1

##### Steps To Reproduce
1. Observe that the project in the directory before/ can be successfully compiled with `mvn --also-make --batch-mode` without any warnings.
2. Start `code before/project.code-workspace`
3. Navigate to parentB/child3/src/main/java/com/example/Qux.java
4. Put the caret on the declaration of `Bar b`, right-click and select Go to definition.
5. The file Bar.java opens
6. Observe that the project in the directory after/ can be successfully compiled with `mvn --also-make --batch-mode` without any warnings.
7. Start `code after/project.code-workspace`
8. Navigate to parentB/child3/src/main/java/com/example/Qux.java
9. Put the caret on the declaration of `Bar b`.
10. Notice that it now gives the problem "Bar cannot be resolved to a type".

##### Current Result
In step 10, the class Bar cannot be found.

##### Expected Result
Step 10 should be like step 5, that the file Bar.java was opened when Go to definition was invoked.

**Comment by cyberphone** [comment]
Maybe related to https://github.com/redhat-developer/vscode-java/issues/3581

**Comment by RolKau** [comment]
Indeed; thank you very much @cyberphone! Adding the paths as a variable in `.project` and `.classpath` seems to be a work-around. To summarize for other readers, the steps are:

1. Define in `.project` for child3 a `<variable>` tag which refers to `${PROJECT_LOC}/../..` (or whichever relative path outside of the current project source tree you want)

2. Define in `.project` for child3 a `<link>` tag that correlates a location relative to this variable which contains the extra source tree you want to add, with a non-existing directory name, e.g. `child2_src_main_java`.

3. Add in `.classpath` for child3 a `<classpathentry>` tag similar to the one for `src/main/java`, but instead refering to the non-existing directory called `child2_src_main_java` that was created in step 2.

I have updated the example project to illustrate how this can be done in a Maven project.

[issue-3544-update1.zip](https://github.com/redhat-developer/vscode-java/files/15093536/issue-3544-update1.zip)
